lib/resty/kafka/producer.lua

```diff
@@ -353,12 +355,82 @@ local function negotiate_api_version(supported_version_map, api_key, max_support
     return min_version
 end
 
+-- if new arguments already exist in cached producers, then skip creating
+-- may subject to plugin (e.g. "kafka-log") schema updates
+local function _same_cluster(broker_list, producer_config, cluster_name)
+  local same_cluster = true
```

Comment:
let's reverse the logic and default to `false`. Accidentally using a wrong item from the cache may pose a security concern while missing a cache is just inconvenient.

Reply:
Good point.

Reply:
Done.